pkg/operation/botanist/addons.go

```diff
@@ -243,25 +216,8 @@ func (b *Botanist) generateOptionalAddonsChart(_ context.Context) (*chartrendere
 
 	kubernetesDashboardEnabled := gardencorev1beta1helper.KubernetesDashboardEnabled(b.Shoot.GetInfo().Spec.Addons)
 
-	nginxIngressConfig, err := b.GenerateNginxIngressConfig()
-	if err != nil {
-		return nil, err
-	}
-	nginxIngress, err := b.InjectShootShootImages(nginxIngressConfig, images.ImageNameNginxIngressController, images.ImageNameIngressDefaultBackend)
-	if err != nil {
-		return nil, err
-	}
-
 	return b.ShootClientSet.ChartRenderer().Render(filepath.Join(charts.Path, "shoot-addons"), "addons", metav1.NamespaceSystem, map[string]interface{}{
 		"global":               global,
 		"kubernetes-dashboard": common.GenerateAddonConfig(nil, kubernetesDashboardEnabled),
-		"nginx-ingress":        nginxIngress,
```

Comment:
You need to add similar changes for nginx-ingress also other wise resource will be recreated again.
```
"kubernetes-dashboard": common.GenerateAddonConfig(nil, kubernetesDashboardEnabled),
```